### PR TITLE
add libvirt provider support to Vagrant file

### PR DESCRIPTION
**Pull request by krisis**
Issue: https://github.com/heketi/vagrant-heketi/issues/4

Signed-off-by: Krishnan Parthasarathi kparthas@redhat.com

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -6,17 +6,24 @@ NODES = 4
 DISKS = 8
 
 Vagrant.configure("2") do |config|
-    config.vm.box = "chef/centos-7.1"
```
Comment (lpabon) [inline Vagrantfile:9]: Could it work by changing line 9 to:

```
if virtualbox, then 
   config.vm.box = 'chef/...' 
else if libvirt 
   config.vm.box = 'puppet...' 
```

**Comment by lpabon** [comment]
Thanks for the work KP.
1. For the network in the `common` task, you may need to setup a variable which passes to ansible if it is libvirt or virtualbox.
2. You will need to also update the README.md accordingly

File: Vagrantfile (hunk 1 above)
Comment (krisis) [inline Vagrantfile:9]: I am not familiar with ruby. I am managing with just enough syntax learnt from neighbouring code snippets. I have passed an `override` parameter to the provider block as suggested in http://docs.vagrantup.com/v2/providers/configuration.html. I tried the change you suggested only to find Vagrant complain,

```
vm:
* A box must be specified.
```

I don't know how to get past this.

Comment (lpabon) [inline Vagrantfile:9]: Does the vagrant mutate help with this issue?

Comment (krisis) [inline Vagrantfile:9]: Yes. It creates a libvirt-compatible box of the same name. So, we don't
need to override box used based on provider anymore. We still need to
configure the libvirt VMs differently.

Comment (lpabon) [inline Vagrantfile:9]: so do you still need to make changes to the ansible script to make the correct configuration inside the VM?

Comment (krisis) [inline Vagrantfile:9]: Let me check that and get back to you. I didn't test the vagrant-mutate'd
box without skipping setting up of enp0s8 interface.

Comment (krisis) [inline Vagrantfile:9]: If I comment the 'when' clause for "Setting up Network" task in common play, while using libvirt provider, the `vagrant provision` fails with the following error messages,

`
failed: [storage2] => {"changed": true, "cmd": ["ifup", "enp0s8"], "delta": "0:00:00.139595", "end": "2015-07-02 06:49:10.430911", "rc": 1, "start": "2015-07-02 06:49:10.291316", "warnings": []}
stderr: /sbin/ifup: configuration for enp0s8 not found.
`
Conclusion: We need the changes to the ansible script to make it work with libvirt provider.

**Comment by krisis** [comment]
@lpabon did you get a chance to check if this patch doesn't break virtualbox provider?

**Comment by lpabon** [comment]
Verified on virtualbox.

**Pull request merged**